### PR TITLE
refactor: improve listing files from s3

### PR DESCRIPTION
**Pull request by IyadhKhalfallah**
---

- [ ] This PR references an open issue
- [ ] Added a test
- [ ] Updated the documentation
- [ ] Added a screenshot of a successful customer deployment

### CODE REVIEW

**Comment by IyadhKhalfallah** [comment]
Benchmark: 
- **list_objects_v2 (new method):** 47.67947483062744
- **filter (current method):** 53.21679997444153

File: now/utils.py [hunk 1]
```diff
@@ -332,6 +332,29 @@ def get_email():
         return ''
 
 
+def list_s3_objects(bucket_name, prefix, user_input, max_keys=1000):
+    session = boto3.Session(
+        aws_access_key_id=user_input.aws_access_key_id,
+        aws_secret_access_key=user_input.aws_secret_access_key,
+    )
+
+    client = session.client('s3', region_name='eu-west-1')
```
Comment (azayz) [inline now/utils.py:341]: take the region name from user_input

**Review by azayz — CHANGES_REQUESTED** [review]

**Comment by joschkabraun** [comment]
This blogpost details listing 67 billion objects in 169 hours -> so listing 67k objects (lottie) sholud take 0.6 seconds
https://blog.purestorage.com/purely-technical/listing-67-billion-objects-in-1-bucket/

**Comment by IyadhKhalfallah** [comment]
> This blogpost details listing 67 billion objects in 169 hours -> so listing 67k objects sholud take 0.6 seconds https://blog.purestorage.com/purely-technical/listing-67-billion-objects-in-1-bucket/

@joschkabraun I tried that one, and it has actually 2 seconds more latency than the one in the PR. Something that might make this better, as mentioned in the blogpost, is using asyncio and ProcessPoolExecutor to concurrently list each keys. That will actually interfere with the existing event loop that we have for the prompt. Do you think we should give this a shot? If yes, should this be part of this ticket?

**Review by azayz — APPROVED** [review]
LGTM after Joschka's comment

**Pull request closed without merging**